docs/rfcs/header-component-api.md

```diff
@@ -0,0 +1,418 @@
+        href="https://www.theguardian.com/world"
+    />
+</Navigation.SecondaryLinksGroup>
+```
+
+###### `<Navigation.SecondaryLink />`
+
+A navigational link that appears in the `Navigation.SecondaryLinks` component
+
+![Navigation secondary link](images/header-component-api/navigation-secondary-link.png)
+
+###### Props
+
+-   `href` _string_
+    -   navigation location when `Navigation.SecondaryLink` is clicked
+-   `label` _string_
+    -   display label of the `Navigation.SecondaryLink`
+
+###### Example
+
+```jsx
+<Navigation.SecondaryLink
+    label="World news"
+    href="https://www.theguardian.com/world"
+/>
+```
+
+##### `<Navigation.SecondaryLinksSupplementalLink />`
+
+A visually prominent navigational link that appears in the `Navigation.SecondaryLinks` component.
+
+![Navigation secondary links supplemental link](images/header-component-api/navigation-secondary-links-supplemental-link.png)
+
+###### Example
+
+```jsx
+<Navigation.SecondaryLinksSupplementalLink
+    label="Masterclasses"
+    href="https://www.theguardian.com/guardian-masterclasses"
+/>
+```
+
+###### Props
+
+-   `href` _string_
+    -   navigation location when `Navigation.SecondaryLinksSupplementalLink` is clicked
+-   `label` _string_
+    -   display label of the `Navigation.SecondaryLinksSupplementalLink`
+
+## Specialised headers
+
+### `<EditorialHeader isSignedIn showSubscriberMessage edition />`
+
+A special header layout for editorial pages. The logo and supporter messages are hard-coded.
+
+![Editorial header](images/header-component-api/editorial-header.png)
+
+#### Differences to generic header
+
+##### Brandbar
+
+-   Links to the left, floated right
+-   Editions dropdown to the right
+-   My account / sign in
+    -   Links hierarchy
+    -   Primary link – always visible
+    -   Secondary links – visible at wider breakpoints
+    -   Editions menu
+    -   Is this a concept we want to reflect in the API?
+-   Subscribe / Contribute links
+-   `<SubscribeMessage />`?
+    -   Hardcoded in Source?
+    -   Currently inserted by automat
+
+#### Example
+
+```jsx
+<EditorialHeader isSignedIn showMessage>
+    <BrandBar>
+        <BrandBar.Links>
+            <BrandBar.Link href icon={<Icon />} label until from />
+            <BrandBar.LinkList icon label until from>
+                <BrandBar.Link href label />
+                <BrandBar.Link href label />
+            </BrandBar.LinkList>
+            <BrandBar.Link href icon={<Icon />} label until from />
+        </BrandBar.Links>
+        <EditorialHeader.EditionSwitch onChange>
+            <EditorialHeader.EditionSwitchItem href label />
+            <EditorialHeader.EditionSwitchItem href label selected />
+            <EditorialHeader.EditionSwitchItem href label />
+        </EditorialHeader.EditionSwitch>
+    </BrandBar>
+    <Navigation />
+    <Subnav />
+</EditorialHeader>
+```
+
+#### Props
+
+-   `isSignedIn` _boolean_
+    -   whether to show the support buttons
```

Comment:
This is determined by a call to the identity API made client-side. If we pass it in as a prop, it will trigger a re-render of the whole header, which might not be ideal.